### PR TITLE
De-static IceAdapter class

### PR DESCRIPTION
**Pull request by Brutus5000**
Prepare a slow transition from static-everything to dependency injection

Stripped out of #38

### CODE REVIEW

File: ice-adapter/src/main/java/com/faforever/iceadapter/debug/DebugWindow.java [hunk 1]
```diff
@@ -99,10 +99,10 @@ public void initStaticVariables() {
 
         runOnUIThread(() -> {
             controller.versionLabel.setText("Version: %s".formatted(IceAdapter.VERSION));
```
Comment (GodFuper) [inline ice-adapter/src/main/java/com/faforever/iceadapter/debug/DebugWindow.java:101]: why not IceAdapter.version()

File: ice-adapter/src/main/java/com/faforever/iceadapter/rpc/RPCService.java [hunk 2]
```diff
@@ -24,22 +24,16 @@
...
     private static volatile boolean skipRPCMessages = false;
 
-    static {
-        objectMapper.registerModule(new JavaTimeModule());
-    }
-
-    public static void init() {
-        log.info("Creating RPC server on port {}", IceAdapter.RPC_PORT);
+    public static void init(int port) {
+        Debug.RPC_PORT = port;
```
Comment (GodFuper) [inline ice-adapter/src/main/java/com/faforever/iceadapter/rpc/RPCService.java:32]: Why RPC_PORT not in RPCService?

File: ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java [hunk 3]
```diff
@@ -20,62 +21,41 @@
...
 
-    public static void start(IceOptions iceOptions) {
+    public void start() {
         determineVersion();
+        log.info("Version: {}", VERSION);
 
-        loadOptions(iceOptions);
+        Debug.DELAY_UI_MS = iceOptions.getDelayUi();
+        Debug.ENABLE_DEBUG_WINDOW = iceOptions.isDebugWindow();
+        Debug.ENABLE_INFO_WINDOW = iceOptions.isInfoWindow();
```
Comment (GodFuper) [inline ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java:51]: Why do we create a new static variable if we have it in IceOptions?

File: ice-adapter/src/main/java/com/faforever/iceadapter/rpc/RPCService.java (hunk 2 above)
Comment (Brutus5000) [reply ice-adapter/src/main/java/com/faforever/iceadapter/rpc/RPCService.java:32]: same as above

File: ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java (hunk 3 above)
Comment (Brutus5000) [reply ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java:51]: Will come in a later commit ;) One thing at a time.

**Comment by Brutus5000** [comment]
@GodFuper Okay I made all static variables of IceAdapter class private now, with restricted access via (still static) getters.

If you look into the original PR, I try to go one class at a time, so the review scope is smaller and I don't intentionally introduce different behavior.

**Pull request merged**